From 5e87f17f191018784ac90092a6131cb4ddf265f0 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Tue, 4 Sep 2018 21:31:59 +0800
Subject: [PATCH 221/706] arm64: dts: ls1028a: Add Mali DP-500 support

This patch adds Mali DP-500 dts support for LS1028AQDS/RDB/SIMU/PXP
boards.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 24285eae5b62122627aef857266356d8bc180368)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-pxp.dts    |  4 +++
 .../boot/dts/freescale/fsl-ls1028a-qds.dts    |  4 +++
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts    |  4 +++
 .../boot/dts/freescale/fsl-ls1028a-simu.dts   |  4 +++
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 33 +++++++++++++++++++
 5 files changed, 49 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
index ca49b07c0bc8..944c247a4594 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
@@ -55,3 +55,7 @@
 	aliases {
 	};
 };
+
+&dp0 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
index 8647195e300e..961ba4c4ab2e 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
@@ -107,6 +107,10 @@
 	};
 };
 
+&dp0 {
+	status = "okay";
+};
+
 &dspi0 {
 	bus-num = <0>;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 35aa2e130093..e67df19154a5 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -222,3 +222,7 @@
 &sai4 {
 	status = "okay";
 };
+
+&dp0 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
index b94f940e6b40..ae4b879afe02 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
@@ -55,3 +55,7 @@
 	aliases {
 	};
 };
+
+&dp0 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 4eb1317c8e1e..f25c8981c10e 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -109,6 +109,27 @@
 		clock-output-names = "sysclk";
 	};
 
+	dpclk: dpclk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+		clock-output-names= "dpclk";
+	};
+
+	aclk: aclk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <650000000>;
+		clock-output-names= "aclk";
+	};
+
+	pclk: pclk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <325000000>;
+		clock-output-names= "pclk";
+	};
+
 	reboot {
 		compatible ="syscon-reboot";
 		regmap = <&dcfg>;
@@ -577,5 +598,17 @@
 			status = "disabled";
 		};
 	};
+
+	dp0: malidp@f080000 {
+		compatible = "arm,mali-dp500";
+		reg = <0x0 0xf080000 0x0 0x10000>;
+		interrupts = <0 222 0x4>,
+			     <0 223 0x4>;
+		interrupt-names = "DE", "SE";
+		clocks = <&dpclk>, <&aclk>,
+			 <&aclk>, <&pclk>;
+		clock-names = "pxlclk", "mclk", "aclk", "pclk";
+		arm,malidp-output-port-lines = /bits/ 8 <8 8 8>;
+	};
 };
 
-- 
2.17.1

